image-sync/image_sync/sync.py

```diff
@@ -58,14 +73,55 @@ def fetch_site_info():
         }
     """
     params = {"query": query}
-    r = requests.get(
+    r = httpx.get(
         CONF.sync.graphql_url, params=params, headers={"accept": "application/json"}
     )
     r.raise_for_status()
     data = r.json()["data"]["siteCloudComputingEndpoints"]["items"]
     return data
 
 
+def fetch_site_info_cloud_info():
+    logging.debug("Fetching site info from cloud-info")
+    sites = []
+    # 1 - Get all sites listing
+    r = httpx.get(CONF.sync.cloud_info_url, headers={"Accept": "application/json"})
+    r.raise_for_status()
+    # 2 - Go one by one getting the shares
+    for file in r.json():
+        try:
+            r = httpx.get(os.path.join(CONF.sync.cloud_info_url, file["name"]))
+            r.raise_for_status()
+        except httpx.HTTPError as e:
+            logging.warning(f"Exception while trying to get {file['name']}: {e}")
+            continue
+        full_site = r.json()
+        admin_domain = full_site["CloudComputingService"][0]["Associations"].get(
+            "AdminDomain", None
+        )
+        if not admin_domain:
+            continue
+        # Some associations are a list, other directly strings, cloud-info should harmonise
+        site = admin_domain[0]
```

Comment:
Does this not run into problems with `admin_domain` is `None` from above?

Reply:
If `None`, then site basically will be ignored (`continue` in line 103). Although it's true that if the admin_domain is not a list, this would fail or give a wrong value. 
We should monitor if there are any failures and adapt the code here or in cloud-info-provide